auth/service.cc

```diff
@@ -131,7 +131,8 @@ service::service(
         ::service::migration_notifier& mn,
```

Comment:
Re ` auth/service: don't create system_auth keyspace when used by maintenance socket` commit message

> The maintenance socket is created before joining the cluster. When an auth service
is started it creates system_auth keyspace if it's missing. It is not synchronized
with other nodes, because this node hasn't joined the group0 yet. Thus a node has
a mismatched schema and is unable to join the cluster.

I don't understand this description -- it's written as if it was describing current state of Scylla. But obviously nodes are able to join clusters and system_auth is created using group 0?

I think some context is missing.

Reply:
It would be a bit better if before mentioning "maintenance auth service" you introduced the concept. The first paragraph assumes that the reader knows what "maintenance auth service" is, but that concept is only introduced in the last paragraph.

The best order would be:
- explain that we want to create a separate auth service for maintenance socket (and why)
- then explain that there's a problem with that -- this happens too early and therefore blah blah
- finally the solution

But it's minor, the current commit message is good enough.